react/index.d.ts

```diff
@@ -14,7 +14,7 @@ declare namespace React {
 
     type ReactType = string | ComponentClass<any> | StatelessComponent<any>;
 
-    type Key = string | number;
+    type Key = string | number | null;
```

Comment:
> the `Key` value can actually be null from the built in methods

Great, that's what I'm asking about. Did you consider changing the `key` property on `interface ReactElement<P>` to `key: Key | null` instead?